src/pybind/mgr/zabbix/module.py

```diff
@@ -269,30 +269,35 @@ def send(self):
             })
             return
 
-        try:
-            self.log.info(
-                'Sending data to Zabbix server %s as host/identifier %s',
-                self.config['zabbix_host'], identifier)
-            self.log.debug(data)
+        servers = self.config['zabbix_host'].split(",")
+        result = True
 
-            zabbix = ZabbixSender(self.config['zabbix_sender'],
-                                  self.config['zabbix_host'],
-                                  self.config['zabbix_port'], self.log)
+        for server in servers:
+            zabbix_host, sep, zabbix_port = server.partition(':')
```

Comment:
How will this handle IPv6 addresses? The current implementation supports IPv6 addresses as Zabbix host, but this will probably break with this code as it will split on ':' which is present in an IPv6 address.

How can we handle that?

Reply:
In addition, I checked the current code and that validates if the port number which is provided is a valid integer.

This code simply assumes that everything after the semi-column is a port(number).

I think we could need to have a validator on the string being passed as Zabbix Server to have this done properly.

On module init we should then already create the in-memory list/dict with the Zabbix servers instead of just splitting the string every time we do a run.

Reply:
OK, makes sense. I will try to rework it.